nix/tools/withTools.nix

```diff
@@ -74,19 +76,50 @@ let
           pg_ctl -l "$tmpdir/db.log" -w start -o "-F -c listen_addresses=\"\" -c hba_file=$HBA_FILE -k $PGHOST -c log_statement=\"all\" " \
             >> "$setuplog"
 
+          log "Creating a minimally privileged $PGUSER connection role..."
+          createuser "$PGUSER" -U postgres --host="$tmpdir/socket" --no-createdb --no-inherit --no-superuser --no-createrole --no-replication --login
+
+          >&2 echo "${commandName}: You can connect with: psql 'postgres:///$PGDATABASE?host=$PGHOST' -U postgres"
+          >&2 echo "${commandName}: You can tail the logs with: tail -f $tmpdir/db.log"
+
+          if test "$_arg_replica" = "on"; then
+            replica_slot="replica_$RANDOM"
+            replica_dir="$tmpdir/$replica_slot"
+            replica_host="$tmpdir/socket_$replica_slot"
+
+            mkdir -p "$replica_host"
+
+            replica_dblog="$tmpdir/db_$replica_slot.log"
+
+            log "Running pg_basebackup for $replica_slot"
+
+            pg_basebackup -v -h "$PGHOST" -U postgres --wal-method=stream --create-slot --slot="$replica_slot" --write-recovery-conf -D "$replica_dir" \
+              >> "$setuplog" 2>&1
```

Comment:
The replica process goes to the same `$setuplog`. I had another one previously, but seemed better to have it all on one file since `$setuplog` is already light on details.